src/metric-engine/src/engine/create.rs

```diff
@@ -363,6 +363,12 @@ impl MetricEngineInner {
                 column: DATA_SCHEMA_TSID_COLUMN_NAME,
             }
         );
+        ensure!(
+            !name_to_index.contains_key(PRIMARY_KEY_COLUMN_NAME),
+            InternalColumnOccupiedSnafu {
+                column: PRIMARY_KEY_COLUMN_NAME,
+            }
+        );
```

Comment:
The mito engine will check this if you use the existing internal column. We have a method `is_internal_column()`. You don't need to check this manually.